### PR TITLE
feat(Anix): add presence

### LINKED ISSUE #7519: Anix | anix.to
### Website name

Anix

### Website URL

https://anix.to/

### Website logo

https://cdn.discordapp.com/attachments/1069829519000744027/1132862867708850176/logo.png

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

Same thing like any other anime website presence.

### PR DESCRIPTION
**Pull request by vahya**
## Description 

Closes #7519 

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

</details>

### CODE REVIEW

File: websites/A/Anix/metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/gtqJrml.png",
+	"thumbnail": "https://i.imgur.com/u2ZMJZV.png",
+	"color": "#95b6d0",
+	"category": "anime",
+	"tags": [
+		"anime"
+	],
+	"iframe": true,
+	"iFrameRegExp": ".*",
```
Comment (theusaf) [inline websites/A/Anix/metadata.json:24]: There are new fields called `matches` and `iFrameMatches`, which are required for new Presences.
These are an array of strings which need to follow https://developer.chrome.com/docs/extensions/develop/concepts/match-patterns.

File: websites/A/Anix/metadata.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	"service": "Anix",
+	"description": {
+		"en": "Anix offers a freemium anime streaming experience with premium features, allowing you to watch anime online for free without registration."
+	},
+	"url": [
+		"anix.vc",
+		"anix.to"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/gtqJrml.png",
```
Comment (theusaf) [inline websites/A/Anix/metadata.json:16]: [Logos](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo) must be 512x512px

File: websites/A/Anix/metadata.json (hunk 1 above)
Comment (theusaf) [inline websites/A/Anix/metadata.json:24]: If possible, change the `iFrameRegExp` to be more specific if you can determine all iframe hosts the website uses.

File: websites/A/Anix/metadata.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	"thumbnail": "https://i.imgur.com/u2ZMJZV.png",
+	"color": "#95b6d0",
+	"category": "anime",
+	"tags": [
+		"anime"
+	],
+	"iframe": true,
+	"iFrameRegExp": ".*",
+	"settings": [
```
Comment (theusaf) [inline websites/A/Anix/metadata.json:25]: Please also add a [screenshot of the settings](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#reviewing) to your PR description

File: websites/A/Anix/presence.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
+const presence = new Presence({
+		clientId: "1222296154058916013",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
+	Logo = "logo",
```
Comment (theusaf) [inline websites/A/Anix/presence.ts:7]: Use an image URL rather than a Discord asset name.

File: websites/A/Anix/presence.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			details: "Browsing",
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, href, search } = document.location,
+		pathArr = pathname.split("/"),
+		[buttons] = await Promise.all([presence.getSetting<boolean>("buttons")]);
```
Comment (theusaf) [inline websites/A/Anix/presence.ts:35]: As you are only reading one setting, `Promise.all` is not needed.

File: websites/A/Anix/presence.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+				.querySelector(".sidebar-set > .wrap > a.active")
+				.textContent.trim();
+
+			if (presenceData.details === "Watch List") {
+				presenceData.state = document.querySelector(
+					".user-folder.active"
+				).textContent;
+			}
+
+			break;
```
Comment (theusaf) [inline websites/A/Anix/presence.ts:104]:
```suggestion
			}
			break;
```

File: websites/A/Anix/presence.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
+const presence = new Presence({
+		clientId: "1222296154058916013",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
```
Comment (theusaf) [inline websites/A/Anix/presence.ts:6]:
```suggestion
const enum Assets {
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**